clustered/client/src/main/java/org/ehcache/clustered/client/internal/service/ValueCodecFactory.java

```diff
@@ -79,10 +88,14 @@ public T decode(Object input) {
       ValueWrapper data = (ValueWrapper) input;
       ByteArrayInputStream bais = new ByteArrayInputStream(data.getValue());
       try {
-        try (ObjectInputStream ois = new ObjectInputStream(bais)) {
+        try (ObjectInputStream ois =
+               (whiteListedClasses != null) ? new WhiteListObjectInputStream(bais, whiteListedClasses)
```

Comment:
This null special casing is a bit ugly.  Would it be easier to write a "contains-all" set implementation?  You could even just have a Predicate instead of an explicit whitelist.

Reply:
Are you saying to push this part inside WhitelistObjectInputstream, yes null handling is ugly, did not get the alternative you are proposing here

Reply:
Either:
1. Switching to passing a predicate to the WhiteListObjectInputStream.  When whitelisting this would be `whitelist::contains` and when not whitelisting it would be `c -> true`.
2. Writing a "contains-eeverything set implementation, and passing that in when not whitelisting.

Personally I prefer option 1.

Reply:
Done, taking in Predicate<Class<?>> as input parameter now